common/txmgr/types/client.go

```diff
@@ -3,38 +3,40 @@ package types
 import (
 	"context"
 	"math/big"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
 )
 
 // A generic client interface for communication with the RPC node
 // Every native chain must implement independently
-type Client[CHAINID any, ADDR comparable, BLOCK any, TX any, HASH comparable, TXRECEIPT any, EVENT any, EVENTOPS any] interface {
+type Client[CHAINID ID, SEQ Sequence, ADDR types.Hashable[ADDR], BLOCK any, TX any, HASH types.Hashable[HASH], TXRECEIPT any, EVENT any, EVENTOPS any] interface {
```

Comment:
nit: Use CHAIN_ID, as that's the name used in other places too.

Reply:
Also, TX_HASH instead of HASH. That helps differentiate if this is Tx hash, or BlockHash.

Reply:
Why do we use different hash generics for tx and block? Are there known cases where it's different?

Reply:
If we start using CHAIN_ID, then we'd have to be consistent with other namings, e.g. TX_RECEIPT, TX_HASH, TX_ATTEMPT, EVENT_OPS etc. So far I see the mixed usage everywhere, let's stick to either of naming convention then.